src/middlewared/middlewared/plugins/account.py

```diff
@@ -1022,9 +1020,23 @@ def get_user_obj(self, data):
             user_obj['grouplist'] = None
 
         if data['sid_info']:
+            sid = None
             match user_obj['source']:
-                case 'LOCAL' | 'ACTIVEDIRECTORY':
-                    # winbind provides idmapping for local and AD users
+                case 'LOCAL':
+                    idmap_ctx = None
+                    db_entry = self.middleware.call_sync('user.query', [[
+                        'username', '=', user_obj['pw_name']
+                    ]], {'select': ['sid']})
+                    if not db_entry:
+                        self.logger.error(
+                            '%s: local user exists on server but does not exist in the '
+                            'the user account table.', user_obj['pw_name']
+                        )
+                        sid = None
```

Comment:
`sid` is now already set to `None` above, so this line is unnecessary.  (But meh.)